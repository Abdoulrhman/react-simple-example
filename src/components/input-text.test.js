import React from 'react';
import InputText from './input-text';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


describe('InputText Component', () => {
  let component;
  let props;
  let label = 'keywords';
  let keywords = 'this is some keyword';

  Enzyme.configure({
    adapter: new Adapter()
  });

  beforeEach(() => {
    props = {
      onChange: jest.fn()
    }
    
    component = shallow(
      <InputText 
        label={label}
        value={keywords}
        onChange={props.onChange} />
    );
  })

  it('Should render the right way', ()=> {
    expect(component).toMatchSnapshot();
  })
})