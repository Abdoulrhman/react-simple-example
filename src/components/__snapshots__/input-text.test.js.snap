// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText Component Should render the right way 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputText
    label="keywords"
    onChange={[MockFunction]}
    value="this is some keyword"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          keywords
          :
        </label>,
        <input
          className="form-control"
          onChange={[Function]}
          type="text"
          value="this is some keyword"
        />,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "keywords",
            ":",
          ],
        },
        "ref": null,
        "rendered": Array [
          "keywords",
          ":",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control",
          "id": undefined,
          "onChange": [Function],
          "type": "text",
          "value": "this is some keyword",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            keywords
            :
          </label>,
          <input
            className="form-control"
            onChange={[Function]}
            type="text"
            value="this is some keyword"
          />,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "keywords",
              ":",
            ],
          },
          "ref": null,
          "rendered": Array [
            "keywords",
            ":",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "id": undefined,
            "onChange": [Function],
            "type": "text",
            "value": "this is some keyword",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
